src/workerd/jsg/value.h

```diff
@@ -1048,10 +1077,15 @@ class DictWrapper {
           throwTypeError(context->GetIsolate(), TypeErrorContext::dictField(strName.cStr()),
               TypeWrapper::getName((V*)nullptr));
         }
-        builder.add(typename Dict<V, K>::Field{
-          KJ_ASSERT_NONNULL(kj::mv(unwrappedName)), KJ_ASSERT_NONNULL(kj::mv(unwrappedValue))});
+        entries.upsert(
+            KJ_ASSERT_NONNULL(kj::mv(unwrappedName)), KJ_ASSERT_NONNULL(kj::mv(unwrappedValue)));
```

Comment:
I'm not sure if silently ignoring the potentially duplicate entries makes sense and is what we'd want.

Reply:
It's what URLSearchParams is expected to do; I could move the behaviour over there if it makes more sense.

Reply:
Moved to the caller